File: back/src/main/java/com/openclassrooms/mddapi/payload/request/CommentRequest.java
package com.openclassrooms.mddapi.payload.request;

import jakarta.validation.constraints.NotBlank;

/**
 * Represents a request payload for creating or updating a comment.
 * This class encapsulates the data needed to create or modify a comment.
 */
public class CommentRequest {

    /**
     * The content of the comment.
     * This field cannot be blank.
     */
    @NotBlank
    private String content;

    /**
     * The identifier of the article to which the comment belongs.
     * This field cannot be blank.
     */
    @NotBlank
    private Integer article_id;

    /**
     * Gets the content of the comment.
     * 
     * @return the content of the comment
     */
    public String getContent() {
        return content;
    }

    /**
     * Sets the content of the comment.
     * 
     * @param content the content of the comment
     */
    public void setContent(String content) {
        this.content = content;
    }

    /**
     * Gets the identifier of the article to which the comment belongs.
     * 
     * @return the article identifier
     */
    public Integer getArticle_id() {
        return article_id;
    }

    /**
     * Sets the identifier of the article to which the comment belongs.
     * 
     * @param article_id the article identifier
     */
    public void setArticle_id(Integer article_id) {
        this.article_id = article_id;
    }
}

File: front/src/app/features/articles/interfaces/comment-request.ts
/**
 * Interface representing a request to create a new comment.
 * 
 * The `CommentRequest` interface defines the structure of the data required to create
 * a new comment on an article. It includes the content of the comment and the identifier
 * of the article to which the comment is being added.
 * 
 * @interface CommentRequest
 * @property {string} content - The text content of the comment. This field is required.
 * @property {number} article_id - The unique identifier of the article to which the comment belongs. This field is required.
 */
export interface CommentRequest {
    content: string;
    article_id: number;
}
